File: jcv-core/src/main/java/com/ekino/oss/jcv/core/validator/DefaultParameterizedTemplateValidator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator;

import com.ekino.oss.jcv.core.initializer.TemplatedComparatorInitializer;
import org.skyscreamer.jsonassert.ValueMatcher;

public class DefaultParameterizedTemplateValidator<T> extends ValueParameterizedTemplateValidator<T> {

    private final TemplatedComparatorInitializer<T> comparatorInitializer;

    public DefaultParameterizedTemplateValidator(String validatorId, TemplatedComparatorInitializer<T> comparatorInitializer) {
        super(validatorId);
        this.comparatorInitializer = comparatorInitializer;
    }

    @Override
    protected ValueMatcher<T> getValueComparator(ValidatorTemplateManager validatorTemplateManager) {
        return comparatorInitializer.initComparator(validatorTemplateManager);
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/validator/DefaultParameterizedTemplateValidator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator

import com.ekino.oss.jcv.core.initializer.TemplatedComparatorInitializer
import org.skyscreamer.jsonassert.ValueMatcher

class DefaultParameterizedTemplateValidator<T>(
  validatorId: String,
  private val comparatorInitializer: TemplatedComparatorInitializer<T>
) : ValueParameterizedTemplateValidator<T>(validatorId) {

  override fun getValueComparator(validatorTemplateManager: ValidatorTemplateManager): ValueMatcher<T> {
    return comparatorInitializer.initComparator(validatorTemplateManager)
  }
}
